tokio/src/task/local.rs

```diff
@@ -967,7 +975,7 @@ impl Shared {
     fn schedule(&self, task: task::Notified<Arc<Self>>) {
         CURRENT.with(|localdata| {
             match localdata.ctx.get() {
-                Some(cx) if cx.shared.ptr_eq(self) => unsafe {
+                Some(cx) if cx.shared.ptr_eq(self) && !localdata.entered.get() => unsafe {
```

Comment:
it might be worth adding a comment here explaining why we only choose this branch if `entered` is false? although, if we expand the doc comment on `entered`, that could also be fine. really, i just think there ought to be an explicit comment somewhere stating that the `LocalSet` is woken if `entered` is true.